modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/token/RefreshTokenTestCase.java

```diff
@@ -115,74 +106,68 @@ public void testRefreshTokenAPITestCase() throws Exception {
         String tags = "sample, token, media";
         String description = "This is test API create by API manager integration test";
         String apiVersion = "1.0.0";
-
-        //Login to publisher
-        apiPublisher.login(publisherContext.getContextTenant().getContextUser().getUserName(),
-                           publisherContext.getContextTenant().getContextUser().getPassword());
-
         //Create API.
         APIRequest apiRequest = new APIRequest(apiName, apiContext, new URL(backEndEndpointUrl));
         apiRequest.setTags(tags);
         apiRequest.setDescription(description);
         apiRequest.setVersion(apiVersion);
         apiRequest.setSandbox(backEndEndpointUrl);
         apiRequest.setProvider(user.getUserName());
-        apiPublisher.addAPI(apiRequest);
-
+        HttpResponse serviceResponse = restAPIPublisher.addAPI(apiRequest);
         //Publish API.
-        APILifeCycleStateRequest updateRequest =
-                new APILifeCycleStateRequest(apiName, user.getUserName(), APILifeCycleState.PUBLISHED);
-        apiPublisher.changeAPILifeCycleStatus(updateRequest);
-
-        //Login to Store.
-        apiStore.login(storeContext.getContextTenant().getContextUser().getUserName(),
-                       storeContext.getContextTenant().getContextUser().getPassword());
-
-        //Add Application.
-        apiStore.addApplication(APPLICATION_NAME, APIMIntegrationConstants.APPLICATION_TIER.DEFAULT_APP_POLICY_FIFTY_REQ_PER_MIN, "", "this-is-test");
-
-        //Subscribe Application to API.
-        SubscriptionRequest subscriptionRequest = new SubscriptionRequest(apiName, storeContext.getContextTenant()
-                .getContextUser()
-                .getUserName());
-        subscriptionRequest.setTier(APIMIntegrationConstants.API_TIER.GOLD);
-        subscriptionRequest.setApplicationName(APPLICATION_NAME);
-        apiStore.subscribe(subscriptionRequest);
+        apiId = serviceResponse.getData();
+        restAPIPublisher.changeAPILifeCycleStatus(apiId, Constants.PUBLISHED);
+        String gatewayUrl = getAPIInvocationURLHttp("tokenTestAPI/1.0.0/customers/123");
+        // Add application
+        ApplicationDTO applicationDTO = restAPIStore.addApplicationWithTokenType("TokenTestAPI-Application",
+                APIMIntegrationConstants.APPLICATION_TIER.UNLIMITED, "", "this-is-test",
+                "OAUTH");
+        tokenTestApiAppId = applicationDTO.getApplicationId();
+        SubscriptionDTO subscriptionDTO = restAPIStore.subscribeToAPI(apiId, applicationDTO.getApplicationId(),
+                APIMIntegrationConstants.API_TIER.GOLD);
+        assertEquals(true, subscriptionDTO.getThrottlingPolicy().equals("Gold"));
```

Comment:
Shall we use the same constant `APIMIntegrationConstants.API_TIER.GOLD` here too?

Reply:
fixed